packages/styled-system/src/sort.ts

```diff
@@ -0,0 +1,34 @@
+import { Dict } from "@chakra-ui/utils"
+
+const collator = new Intl.Collator(undefined, {
+  numeric: true,
+  sensitivity: "base",
+})
+
+const cssOrder = ["&", "@media", "&:hover", "&:active", "&:focus"]
+
+function findLastIndex<T>(
+  haystack: T[],
+  predicate: (value: T, index: number, obj: T[]) => unknown,
+) {
+  const reversedIndex = haystack.slice().reverse().findIndex(predicate)
+  const maxIndex = haystack.length - 1
+  return reversedIndex < 0 ? reversedIndex : maxIndex - reversedIndex
+}
+
+function byOrder([keyA]: [string, any], [keyB]: [string, any]) {
+  const aSortIndex = findLastIndex(cssOrder, (key) => keyA.startsWith(key))
+  const bSortIndex = findLastIndex(cssOrder, (key) => keyB.startsWith(key))
+  const aOrder = aSortIndex === -1 ? Number.MAX_SAFE_INTEGER : aSortIndex
+  const bOrder = bSortIndex === -1 ? Number.MAX_SAFE_INTEGER : bSortIndex
+
+  const indexDiff = aOrder - bOrder
+  if (indexDiff !== 0) return indexDiff
+  return collator.compare(keyA, keyB)
```

Comment:
@segunadebayo I think the alphabetical sorting leads to some unexpected side effects. Please compare the IconButton in this branch and `main`. 

I think we should return here `0` and mention in our docs that in an emotion css object the order of properties behave like CSS properties -> the order matters, if they can override one another like `padding` and `padding-left`

Reply:
Got it! I'll take a look at this.

Reply:
@TimKolberger. I agree! This leads to unexpected behavior. I'll return `0` instead.